lighthouse-core/audits/dobetterweb/no-mutation-events.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audit a page to see if it is using Mutation Events (and suggest
+ *     MutationObservers instead).
+ */
+
+'use strict';
+
+const url = require('url');
+const Audit = require('../audit');
+const Formatter = require('../../formatters/formatter');
+
+const MUTATION_EVENTS = [
+  'DOMAttrModified',
+  'DOMAttributeNameChanged',
+  'DOMCharacterDataModified',
+  'DOMElementNameChanged',
+  'DOMNodeInserted',
+  'DOMNodeInsertedIntoDocument',
+  'DOMNodeRemoved',
+  'DOMNodeRemovedFromDocument',
+  'DOMSubtreeModified'
+];
+
+class NoMutationEventsAudit extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'JavaScript',
+      name: 'no-mutation-events',
+      description: 'Site does not use Mutation Events in its own scripts',
+      helpText: 'Using <a href="https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Mutation_events" target="_blank">Mutation events</a> degrades application performance. They are deprecated in the DOM events spec, replaced by <a href="https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver" target="_blank">MutationObservers</a>.',
+      requiredArtifacts: ['URL', 'PageLevelEventListeners']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    if (typeof artifacts.PageLevelEventListeners === 'undefined' ||
+        artifacts.PageLevelEventListeners === -1) {
+      return NoMutationEventsAudit.generateAuditResult({
+        rawValue: -1,
+        debugString: 'PageLevelEventListeners gatherer did not run'
+      });
+    } else if (artifacts.PageLevelEventListeners.rawValue === -1) {
+      return NoMutationEventsAudit.generateAuditResult(artifacts.PageLevelEventListeners);
+    }
+
+    const listeners = artifacts.PageLevelEventListeners;
```

Comment:
You got it. Capturing the improvements in https://github.com/GoogleChrome/lighthouse/issues/858